tests/structures/test_generator.py

```diff
@@ -71,28 +71,158 @@ def fizz_buzz(self):
             for i in myinterview.fizz_buzz():
                 print(i)
             """)
-    
+
     def test_yield_from_not_used(self):
         """Yield from is currently not implemented.
-        Unused yield from statements must not break 
+        Unused yield from statements must not break
         the program compilation, but only ensue a warning."""
         self.assertCodeExecution("""
-            
             def unused():
                 yield from range(5)
 
-            print('Hello, world!')
+                print('Hello, world!')
             """)
-    
+
     @expectedFailure
     def test_yield_from_used(self):
         """Yield from is currently not implemented.
         Yield from statements become NotImplementedErrors at runtime."""
         self.assertCodeExecution("""
-            
             def using_yieldfrom():
                 yield from range(5)
-            
+
             for i in using_yieldfrom():
                 print(i)
             """)
+
+    def test_generator_send(self):
+        self.assertCodeExecution("""
+            def gen():
+                a = yield
+                print(a)
+
+            g = gen()
+            next(g)
+            try:
+                g.send("Hello World")
+            except StopIteration:
+                pass
+            """)
+
+    def test_generator_multi_send(self):
+        self.assertCodeExecution("""
+            def gen():
+                a = yield 1
+                print(a)
+                b = yield 2
+                print(b)
+
+            g = gen()
+            next(g)
+            try:
+                print(g.send("a"))
+                print(g.send("b"))
+            except StopIteration:
+                pass
+            """)
+
+    def test_generator_power_generator(self):
+        self.assertCodeExecution("""
+            def gen():
+                while True:
+                    x = yield 1
+                    yield x ** 2
```

Comment:
Is there any particular reason why a "power" operator is being used here? Is it just a mathematical operator to modify the value a bit? In terms of test naming, it seems like there is some sort of deeper significance to the choice of power specifically.

Reply:
I noticed pow is handled differently in `visit_BinOp` so I added this test case to see its behavior, it executes as intended and I have merged this test case under `test_generator_yield_expr_binary_op`